framework/src/modules/chain/transport.js

```diff
@@ -223,80 +223,35 @@ class Transport {
 	attachSharedMethods() {
 		return {
 			/**
-			 * Description of blocksCommon.
-			 *
-			 * @todo Add @param tags
-			 * @todo Add @returns tag
-			 * @todo Add description of the function
+			 * Returns the common blocks between query.ids and the database blocks table
+			 * @param query
+			 * @param query.ids
+			 * @return {Promise<Array<String>>}
 			 */
-			async blocksCommon(query) {
-				query = query || {};
-
+			async getCommonBlocks(query) {
 				const valid = library.schema.validate(
 					query,
-					definitions.WSBlocksCommonRequest
+					definitions.WSGetCommonBlocksRequest
 				);
 
 				if (!valid) {
 					const err = library.schema.getLastErrors();
 					const error = `${err[0].message}: ${err[0].path}`;
 					library.logger.debug('Common block request validation failed', {
-						err: error.toString(),
+						err: error,
 						req: query,
 					});
 					throw new Error(error);
 				}
 
-				const escapedIds = query.ids
-					// Remove quotes
-					.replace(/['"]+/g, '')
-					// Separate by comma into an array
-					.split(',')
-					// Reject any non-numeric values
-					.filter(id => /^[0-9]+$/.test(id));
-
-				if (!escapedIds.length) {
-					library.logger.debug('Common block request validation failed', {
-						err: 'ESCAPE',
-						req: query.ids,
-					});
-
-					throw new Error('Invalid block id sequence');
-				}
-
 				try {
-					const row = await library.storage.entities.Block.get({
-						id: escapedIds[0],
-					});
-
-					if (!row.length > 0) {
-						return {
-							success: true,
-							common: null,
-						};
-					}
-
-					const {
-						height,
-						id,
-						previousBlockId: previousBlock,
-						timestamp,
-					} = row[0];
-
-					const parsedRow = {
-						id,
-						height,
-						previousBlock,
-						timestamp,
-					};
-
-					return {
-						success: true,
-						common: parsedRow,
-					};
-				} catch (error) {
-					library.logger.error(error.stack);
-					throw new Error('Failed to get common block');
+					return (await library.storage.entities.Block.get({
+						id_in: query.ids,
+					})).map(commonRawBlock => commonRawBlock.id);
```

Comment:
Probably we should decide on what format we send the response.

Reply:
Please see https://github.com/LiskHQ/lisk-sdk/pull/3816/commits/fb01dd27ff96fa56e495cf970642fcd88cf5c341 's message and let me know what you think.